modules/publicize/publicize.php

```diff
@@ -418,6 +419,87 @@ function save_meta( $post_id, $post ) {
 		// Next up will be ::publicize_post()
 	}
 
+	public function update_published_message( $messages ) {
+		global $post_type, $post_type_object, $post;
+		if ( ! $this->post_type_is_publicizeable( $post_type ) ) {
+			return $messages;
+		}
+		$view_post_link_html = '';
+		$viewable = is_post_type_viewable( $post_type_object );
+		if ( $viewable ) {
+			$view_text = __( 'View post' ); // intentionally omitted domain
+
+			if ( 'jetpack-portfolio' == $post_type ) {
+				$view_text = esc_html__( 'View project', 'jetpack' );
+			}
+
+			$view_post_link_html = sprintf( ' <a href="%1$s">%2$s</a>',
+				esc_url( get_permalink( $post ) ),
+				$view_text
+			);
+		}
+
+		$services = $this->get_publicizing_services( $post->ID );
+		if ( empty( $services ) ) {
+			return $messages;
+		}
+
+		$labels = array();
+		foreach ( $services as $service => $dispay_names ) {
+			$labels[] = sprintf(
+				_x( '%1$s (%2$s)', 'Service name is %1$s, and account name is %2$s.', 'jetpack' ),
+				esc_html( $service ),
+				esc_html( implode( ', ', $dispay_names ) )
+			);
+		}
+
+		$messages['post'][6] = sprintf(
+			_x( 'Post published and sharing on %1$s. %2$s', 'Published Message: services and account string is %1$s, link to view published post is %2$s', 'jetpack' ),
+			implode( ', ', $labels ),
+			$view_post_link_html );
+
+		if ( $post_type == 'post' && class_exists('Jetpack_Subscriptions' ) ) {
+			$subscription = Jetpack_Subscriptions::init();
+			if ( $subscription->should_email_post_to_subscribers( $post ) ) {
+				$messages['post'][6] = sprintf(
+					_x( 'Post published, sending emails to subscribers and sharing post on %1$s. %2$s', 'Published Message: services and account string is %1$s, link to view published post is %2$s', 'jetpack' ),
```

Comment:
Having the different variations of text is not bad for translation. Just the extravariable at the end should be appende outside of the translatable te.xt